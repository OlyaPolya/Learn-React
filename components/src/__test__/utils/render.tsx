import React, { PropsWithChildren } from 'react';
import { render as rtlRender } from '@testing-library/react';
import type { RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { AppStore } from '../../store/slice';

export const render = (
  ui: React.ReactElement,
  { store, ...renderOptions }: Partial<RenderOptions> & { store: AppStore }
) => {
  const Wrapper = ({ children }: PropsWithChildren): JSX.Element => {
    return <Provider store={store}>{children}</Provider>;
  };

  return { store, ...rtlRender(ui, { wrapper: Wrapper, ...renderOptions }) };
};
