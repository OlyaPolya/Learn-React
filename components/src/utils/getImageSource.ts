const IMAGE_URL = 'https://live.staticflickr.com/';

export const getImageSource = (server: string, id: string, secret: string): string => {
  return IMAGE_URL + server + '/' + id + '_' + secret + '.jpg';
};
