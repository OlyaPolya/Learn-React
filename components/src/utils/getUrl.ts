type BaseParams = {
  baseUrl: string;
};

enum QueryKeys {
  TEXT = 'text',
  PAGE = 'page',
  SORT = 'sort',
  PER_PAGE = 'per_page',
}

const URLSearchParams = {
  api_key: 'a878977d69bd49afb645f6ab6eca4065',
  privacy_filter: '1',
  nojsoncallback: '1',
  method: 'flickr.photos.search',
  media: 'photos',
  format: 'json',
  extras: 'description,owner_name,views',
};

const UrlToParamsMap = {
  [QueryKeys.TEXT]: 'searchValue',
  [QueryKeys.PAGE]: 'currentPage',
  [QueryKeys.PER_PAGE]: 'perPage',
  [QueryKeys.SORT]: 'sortType',
};

export const getURL = <T extends BaseParams>(params: T): string => {
  const requestParams = Object.entries(URLSearchParams).reduce((result, [key, value]) => {
    return `${result}${key}=${value}&`;
  }, '');

  return Object.values(QueryKeys).reduce((acc, key) => {
    acc += `&${key}=${
      params[UrlToParamsMap[key as unknown as QueryKeys] as keyof BaseParams] as unknown
    }`;
    return acc;
  }, `${params.baseUrl}${requestParams}`);
};
