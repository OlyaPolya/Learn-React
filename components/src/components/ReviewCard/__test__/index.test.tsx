import React from 'react';
import { render } from '../../../__test__/utils/render';
import ReviewCards from '..';
import fixtures from './fixtures';

describe('<Header/>', () => {
  it('should match snapshot', () => {
    const component = render(<ReviewCards />, { store: fixtures.store });

    expect(component.container).toMatchSnapshot();
  });
});
