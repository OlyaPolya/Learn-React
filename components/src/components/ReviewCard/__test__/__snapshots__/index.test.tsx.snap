// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header/> should match snapshot 1`] = `
<div>
  <div
    class="review-list"
  >
    <div
      class="review-card"
    >
      <h5
        class="review-title"
      >
        name 1
      </h5>
      <span
        class="review-rate"
      >
        1
      </span>
      <img
        alt="review image"
        class="review-image"
        src="src/1"
      />
      <p
        class="review-description__title"
      >
        Review
      </p>
      <p
        class="review-description"
      >
        test description 1
      </p>
    </div>
    <div
      class="review-card"
    >
      <h5
        class="review-title"
      >
        name 2
      </h5>
      <span
        class="review-rate"
      >
        2
      </span>
      <img
        alt="review image"
        class="review-image"
        src="src/2"
      />
      <p
        class="review-description__title"
      >
        Review
      </p>
      <p
        class="review-description"
      >
        test description 2
      </p>
    </div>
    <div
      class="review-card"
    >
      <h5
        class="review-title"
      >
        name 3
      </h5>
      <span
        class="review-rate"
      >
        3
      </span>
      <img
        alt="review image"
        class="review-image"
        src="src/3"
      />
      <p
        class="review-description__title"
      >
        Review
      </p>
      <p
        class="review-description"
      >
        test description 3
      </p>
    </div>
  </div>
</div>
`;
